### ISSUE TITLE
Allow directories to be whitelisted in the .env themes and plugins config.

### ISSUE THREAD

**Issue by whyisjake** [WordPress/gutenberg#21595]
**Describe the bug**
As a user, I would like to config a folder that looks like a common `wp-content` directory. It might have a directory of plugins, themes, mu-plugins, and others. Currently, all themes, and plugins must be declared explicitly. It would be easier to list the common directories.

**To reproduce**
1. Download the [VIP Go Skeleton](https://github.com/Automattic/vip-go-skeleton)
2. Try to [add a config](https://gist.github.com/whyisjake/b87ce2239536c8688e1d563e86066de7) for the common directories.
3. `wp-env start`
4. See error: `Warning: The 'client-mu-plugins' plugin could not be found.`

**Expected behavior**
I would think that the directories would be picked up, with all of the subfolders activating.

**Screenshots**
n/a

**Desktop (please complete the following information):**
 - OS: MacOS
 - Node: v12.16.0

**Comment by noahtallen** [comment]
Conveniently, #21229 will solve most of this problem You will be able to specify:

```json
{
  "plugins": "path/to/folder",
}
```

and then that directory will be mounted _as_ `wp-content/plugins`. Same for themes and for mu-plugins. Looking at the linked file, I think the new format would be:

```json
{
    "plugins": "./plugins",
    "themes": "./themes",
    "mu-plugins": "./mu-plugins",
    "config": {
        "WP_DEBUG_DISPLAY": true
    }
}
```

I guess the only thing that doesn't support is `"./client-mu-plugins/"`. 🤔 

I think we also need to make sure that directories of plugins can be linked. I feel like that could be challenging if WordPress doesn't do it out of the box, though. For example, how do we distinguish a directory that contains plugin source code from a directory that contains directories of plugins? (I could also see that being expensive to compute, possibly?) It still seems like a good thing to support.

**Comment by joemcgill** [comment]
Being able to explicitly load a `wp-content` folder would also allow for a project to test any drop-ins that should be loaded into the local environment.

The way I would hope that this would work is that if a `content` config was specified, that folder would be mounted as the `wp-content` directory, and then any additional plugins or themes specified by the config would _also_ be added so you could pass a configuration along with a project that loaded developer debugging plugins without needing to include them in project itself.

**Comment by noahtallen** [comment]
> and then any additional plugins or themes specified by the config would also be added

Interesting. I see what you mean here. I wonder how will it would work with docker volumes, but it shouldn't be too hard to test.

**Comment by noahtallen** [comment]
This is resolved by #22256, which adds support for mapping any local directory to any location within the WordPress install. Note that wp-env won't do anything with what you map, it just sets it up as a volume on the container (so you will have to make sure whatever you map gets activated)
